frontend/connection/src/DefaultStreamlitEndpoints.ts

```diff
@@ -51,13 +69,66 @@ export class DefaultStreamlitEndpoints implements StreamlitEndpoints {
   public constructor(props: Props) {
     this.getServerUri = props.getServerUri
     this.csrfEnabled = props.csrfEnabled
+    this.sendClientError = props.sendClientError
     this.staticConfigUrl = null
   }
 
   public setStaticConfigUrl(url: string | null): void {
     this.staticConfigUrl = url
   }
 
+  public sendClientErrorToHost(
+    component: string,
+    customComponentName: string,
+    error: string | number,
+    message: string,
+    source: string
+  ): void {
+    this.sendClientError(
+      component,
+      customComponentName,
+      error,
+      message,
+      source
+    )
+  }
+
+  public async checkSourceResponse(
```

Comment:
Might benefit from a comment to explain the purpose of `checkSourceResponse`. It's not immediately obvious here.